### DISCUSSION TITLE
Dynamic content routes generate 404

### DISCUSSION THREAD

**Discussion by retroriff** [nuxt/content#1604] [Q&A] [answered]
I am using i18n and works perfectly when running `yarn dev` but it returns 404 with `yarn generate`. This is my folders structure:

- content/en/component.md
- content/es/component.md

And this is how I load them:

```js
const { data } = await useAsyncData(locale.value, () => {
  return queryContent(locale.value + '/component').findOne()
})
```

I also tried to specify dynamic routes but doesn't work either:

```js
generate: {
    async ready () {
      const { $content } = require('@nuxt/content')
      const files = await $content().only(['slug']).fetch()
      console.log(files)
    }
  }
```
I reproduced an example on [StackBlitz](https://stackblitz.com/edit/github-hpsduz?file=package.json)

**Comment by retroriff** [comment, marked as answer]
The problem was that Content V2 does not work with `SSR: false`.

**Reply by mathijsfr** [reply]
Does it work with any other render modes? It doesn't seem to work for me with prerender, isr or swr either.

**Reply by retroriff** [reply]
Did you try `ssr: true`?

**Reply by mathijsfr** [reply]
Yes, it doesn't work for me.

```
routeRules: {
        '/**': { prerender: true },
        '/blog/**': { prerender: true },
        '/blogs/**': { ssr: true },
        '/dashboard/**': { ssr: false },
        '/account/**': { ssr: false },
        '/change-password/**': { ssr: false },
    }
```

My blog gets loaded fine right now with using nuxt generate. However, blogs is where I list all my blogs with different categories. As soon as I use search I get a 404. I also can't change category inside the blogs/ route, it will fail.
